plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotMetadata.java

```diff
@@ -415,6 +418,44 @@ public Optional<AggregationApplicationResult<ConnectorTableHandle>> applyAggrega
         return Optional.of(new AggregationApplicationResult<>(tableHandle, projections.build(), resultAssignments.build(), ImmutableMap.of(), false));
     }
 
+    private boolean canPushDownAggregateIfGroupingColumnsPresent(ConnectorSession session, Optional<DynamicTable> dynamicTable, List<AggregateFunction> aggregates, Map<String, ColumnHandle> assignments)
+    {
+        if (dynamicTable.isEmpty()) {
+            return true;
+        }
+        List<PinotColumnHandle> groupingColumns = dynamicTable.get().getGroupingColumns();
+        if (groupingColumns.isEmpty()) {
+            return true;
+        }
+        // Either second pass of applyAggregation or dynamic table exists
+        if (aggregates.size() > 1) {
+            return false;
+        }
+        AggregateFunction aggregate = getOnlyElement(aggregates);
```

Comment:
Can `aggregates` be empty at this point?

Reply:
Now aggregates must only contain 1 aggregate.